include/roboteam_ai/analysis/PlayChecker.h

```diff
@@ -12,13 +12,22 @@ namespace rtt::ai::analysis {
 
     class PlayChecker {
     public:
-        PlayChecker() = default;
+        /**
+         * The playchecker gets initialized with some initial plays in the allPlays vector. It also starts with a play in action. On call of the update function it will
+         * check if the current play is still valid. If it is then the execution of this play proceeds as normal. If the play is no longer deemed valid, then
+         * the PlayDecider will recalculate plays that are valid for the world and field state and store them in validPlays, and these plays will then be passed
+         * to the PlayDecider
+         */
+        PlayChecker();
         /**
          * Updates the PlayChecker. When this function is called, we check if the play is still valid for the current gamestate,
          * and if it is not new plays are calculated that are valid for the gamestate.
          */
-        void update(world::World *world, world::Field *field);
+        bool update(world::World *world, world::Field *field);
 
+        const std::vector<std::shared_ptr<rtt::ai::analysis::Play>> &getValidPlays() const;
```

Comment:
you don't need the name spacing here, you are already in namespace rtt::ai::analysis.